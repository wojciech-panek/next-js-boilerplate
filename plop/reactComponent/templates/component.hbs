import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'ramda';

import withIntl from '../../shared/lib/withIntl'; // TODO fix relative import
import { Container } from './{{ camelCase name }}.styles';


export class {{ pascalCase name }} extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <Container>
        {{ pascalCase name }} component
      </Container>
    );
  }
}

export default compose(
  withIntl,
)({{ pascalCase name }});

