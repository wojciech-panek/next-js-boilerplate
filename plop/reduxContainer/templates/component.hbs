import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';


export class {{ pascalCase name }} extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <div>
        {{ pascalCase name }} component
      </div>
    );
  }
}
