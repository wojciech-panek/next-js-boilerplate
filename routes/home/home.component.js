import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import messages from './home.messages';


export class Home extends PureComponent {
  static propTypes = {

  };

  render() {
    return (
      <div>
        Homepage
      </div>
    );
  }
}
