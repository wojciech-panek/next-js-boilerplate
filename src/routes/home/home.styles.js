import styled from 'styled-components';


export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
`;

export const Img = styled.img`
  width: 100px;
`;

