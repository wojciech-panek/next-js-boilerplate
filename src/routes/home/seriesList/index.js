export { default as SeriesList } from './seriesList.container';
