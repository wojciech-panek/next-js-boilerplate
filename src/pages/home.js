import React, { PureComponent } from 'react';

import { Home } from '../routes/home';
import { MaintainersActions } from '../modules/maintainers/maintainers.redux';


export default class HomePage extends PureComponent {
  static async getInitialProps({ isServer, store }) {
    await store.execSagaTasks(isServer, (dispatch) => {
      dispatch(MaintainersActions.fetch());
    });

    return {};
  }

  render() {
    return <Home />;
  }
}
