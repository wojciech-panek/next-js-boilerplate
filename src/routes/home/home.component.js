import React, { PureComponent } from 'react';
import Head from 'next/head';
import { FormattedMessage } from 'react-intl';

import LogoPNG from '../../images/logo.png';
import LogoSVG from '../../images/icons/logo.svg';
import { H1 } from '../../shared/theme/typography';
import { Container, IconsContainer, Img } from './home.styles';
import messages from './home.messages';


export class Home extends PureComponent {
  static propTypes = {

  };

  render() {
    return (
      <Container>
        <FormattedMessage {...messages.pageTitle}>
          {pageTitle => (
            <Head>
              <title key="title">{ pageTitle }</title>
            </Head>
          )}
        </FormattedMessage>

        <H1>
          <FormattedMessage {...messages.welcome} />
        </H1>

        <IconsContainer>

          Example images usage:
          <Img src={LogoPNG} alt="" />
          <LogoSVG style={{ width: '25px' }} />
        </IconsContainer>
      </Container>
    );
  }
}
