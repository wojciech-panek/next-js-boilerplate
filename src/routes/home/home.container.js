import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'ramda';
import { withRouter } from 'next/router';

import withIntl from '../../shared/lib/withIntl';
import { Home } from './home.component';
import { selectLocalesLanguage } from '../../modules/locales/locales.selectors';
import { selectMaintainersItems } from '../../modules/maintainers/maintainers.selectors';


const mapStateToProps = createStructuredSelector({
  language: selectLocalesLanguage,
  maintainers: selectMaintainersItems,
});

export const mapDispatchToProps = (dispatch) => bindActionCreators({
}, dispatch);

export default compose(
  withIntl,
  withRouter,
  connect(mapStateToProps, mapDispatchToProps),
)(Home);
