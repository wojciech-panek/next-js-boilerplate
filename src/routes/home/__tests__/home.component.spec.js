import React from 'react';
import { shallow } from 'enzyme';

import { Home } from '../home.component';


describe('Home: Component', () => {
  const defaultProps = {
  };

  const component = (props) => (
    <Home {...defaultProps} {...props} />
  );

  it('should render Home root', () => {
    const wrapper = shallow(component());
    global.expect(wrapper).toMatchSnapshot();
  });
});
