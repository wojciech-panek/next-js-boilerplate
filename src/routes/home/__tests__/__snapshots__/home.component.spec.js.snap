// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home: Component should render correctly 1`] = `
<homestyles__Container>
  <FormattedMessage
    defaultMessage="Homepage"
    id="home.pageTitle"
    values={Object {}}
  >
    <Component />
  </FormattedMessage>
  <typography__H1>
    <FormattedMessage
      defaultMessage="Hello World!"
      id="home.welcome"
      values={Object {}}
    />
  </typography__H1>
  <homestyles__IconsContainer>
    Example images usage:
    <homestyles__Img
      alt=""
      src="logo.png"
    />
    <LogoSVG
      style={
        Object {
          "width": "25px",
        }
      }
      viewBox="0 0 106 134"
      xmlns="http://www.w3.org/2000/svg"
    />
  </homestyles__IconsContainer>
</homestyles__Container>
`;
